atomix/cluster/src/main/java/io/atomix/raft/protocol/JoinRequest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.atomix.raft.protocol;
+
+import static java.util.Objects.requireNonNull;
+
+import io.atomix.raft.cluster.RaftMember;
+import java.util.Objects;
+
+public final class JoinRequest extends AbstractRaftRequest {
+  private final RaftMember joining;
```

Comment:
Hmm.. `ConfigureRequest` also uses `RaftMember`, this will be difficult to change with out breaking compatibility. :thinking:

Reply:
Since `DefaultRaftMember` is serialized as id + hash + type + updated, we'd only save one int for the hash and one `Instant` for the update timestamp (probably serialized as long?). As you said, the `ConfigureRequest` also does the same and changing it would break backwards compatibility. I think it's fine to continue like this, wdyt?

Reply:
Ya.. that is fine. I didn't notice the transient fields in DefaultRaftMember.